ruby/ql/lib/codeql/ruby/dataflow/internal/DataFlowDispatch.qll

```diff
@@ -1408,6 +1425,11 @@ predicate parameterMatch(ParameterPosition ppos, ArgumentPosition apos) {
   ppos.isAnyNamed() and apos.isKeyword(_)
   or
   apos.isAnyNamed() and ppos.isKeyword(_)
+  or
```

Comment:
Perhaps move these disjuncts up after the `ppos.isSplatAll() and apos.isSplatAll()` case, to keep cases involving splats together.